stmhal/mphalport.c

```diff
@@ -104,15 +104,15 @@ void mp_hal_gpio_clock_enable(GPIO_TypeDef *gpio) {
     } else if (gpio == GPIOH) {
         __GPIOH_CLK_ENABLE();
     #endif
-    #ifdef __GPIOI_CLK_ENABLE
+    #if defined(GPIOI) && defined(__GPIOI_CLK_ENABLE)
```

Comment:
I ran into some of these when doing the M7 port as well.
Upgrading the latest F4 HAL (and changing all of the old defintions to the new ones) would help, then we could get rid of the legacy definitions. But that's also alot of work.